src/audio/module_adapter/module_adapter.c

```diff
@@ -170,9 +170,12 @@ static int module_adapter_dp_queue_prepare(struct comp_dev *dev)
 		size_t min_free_space =
 			sink_get_min_free_space(audio_stream_get_sink(&source_buffer->stream));
 
-		/* create a shadow dp queue */
+		/* create a shadow dp queue
+		 * shadow queue must share runtime_stream_params with the source_buffer
+		 */
 		dp_queue = dp_queue_create(min_available, min_free_space, dp_mode,
-					   buf_get_id(source_buffer));
+					   buf_get_id(source_buffer),
+					   &source_buffer->stream.runtime_stream_params);
```

Comment:
Just to double-check: given the pointer to object is owned outside, how to we handle case where the buffer object is freed? I think module reset is ok, but how about unbind?

Reply:
Good question, and there may be more problematic places... 
The solution would be moving management of shadow dpQueues from "prepare/reset" to "bind/unbind", to keep them always strictly attached to each other.

Reply:
@lgirdwood At the moment we can proceed with this PR, as it won't make anything worse, and is fixing some scenarios
if comp_buffer is freed before dp the system is in big trouble anyway

Reply:
ok with this @marcinszkudlinski . can you add some FIXME/TODO entry and/or a  bug on this?

Reply:
I have a following PR almost ready.

Reply:
@marcinszkudlinski ok, merged you can send next PR on top of this one.